Some compressed videos on Android are not playable on macOS or iOS

### Current behavior

It seems to be a specific problem in the combination of this video and a cell phone model.
On a Xiaomi Redmi Note 10 model phone, this attached video seems to compress well, but it is only playable on Android. On iOS or macOS (QuickTime Player) it cannot be played.
Possibly it is some problem with the codecs, but we don't know how to debug this.

Hundreds of videos are uploaded a week in our app and we had not previously detected compatibility problems.

### Expected behavior

Like many other videos that are compressed on Android phones, it plays on iOS without any problem.


### Platform

- [X] Android
- [X] iOS

### React Native Version
- 0.72.4

### React Native Compressor Version
- 1.8.24

### Reproducible Steps And Demo
- I can't reproduce the error on any of our devices. But we have the output after compress in these devices. There seems to be an error in the format, but I can't understand what the difference is with other videos.
- Original video: https://drive.google.com/file/d/19bX1H31JfZ3gokM3D6fnMLJhQ4UnjGZO/view?usp=drive_link
- Compressed video (not playable in mac OS or iPhone) : https://drive.google.com/file/d/1JLtRUzPIAbz_JnwdVorCftirMcDLyVSH/view?usp=sharing

I have discovered that this is an issue specific to certain videos (such as those captured using WeChat), and it is unrelated to the Android device used for video compression. The problem stems from audio processing. If you set `disableAudio = true`, or use FFmpeg to re-encode the audio of the problematic video, then playback on iOS will proceed without errors.

On iOS, errors similar to the following may be reported:
```
    ICSInfo.cpp:138    maxSFB long exceeds maximum allowed value
    SyntacticElements.cpp:24     Error deserializing ICS info
           AACDecoder.cpp:156    Error in deserializing element
           AACDecoder.cpp:228    Error deserializing packet
  ACMP4AACBaseDecoder.cpp:1438  (0x11786d840) Error decoding packet 3: err = -1, packet length: 139
  ACMP4AACBaseDecoder.cpp:1447  '01 40 22 80 A3 7B B0 85 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2F'
     AudioQueueObject.cpp:2459  Prime: Exiting because mConverterError is 1650549857 (0x1000 req, 0x0 primed)
     AudioQueueObject.cpp:2480  Prime: failed (1650549857); will stop (4096/0 frames)
           AACDecoder.cpp:197    Too few bits left in input buffer
           AACDecoder.cpp:228    Error deserializing packet
  ACMP4AACBaseDecoder.cpp:1438  (0x11786d840) Error decoding packet 4: err = -1, packet length: 139
  ACMP4AACBaseDecoder.cpp:1447  '00 EA 38 09 01 F1 88 74 42 32 88 53 EE 83 59 59 6B DE 22 1E 00 28 14 A4 D0 11 75 98 0C 2A BF 6F DC 40 D3 3D 01 60 00 33 67 DE A2 21 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B C0'
```

I have discovered that this is an issue specific to certain videos (such as those captured using WeChat), and it is unrelated to the Android device used for video compression. The problem stems from audio processing. If you set `disableAudio = true`, or use FFmpeg to re-encode the audio of the problematic video, then playback on iOS will proceed without errors.

On iOS, errors similar to the following reported:
```
    ICSInfo.cpp:138    maxSFB long exceeds maximum allowed value
    SyntacticElements.cpp:24     Error deserializing ICS info
           AACDecoder.cpp:156    Error in deserializing element
           AACDecoder.cpp:228    Error deserializing packet
  ACMP4AACBaseDecoder.cpp:1438  (0x11786d840) Error decoding packet 3: err = -1, packet length: 139
  ACMP4AACBaseDecoder.cpp:1447  '01 40 22 80 A3 7B B0 85 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2D 2F'
     AudioQueueObject.cpp:2459  Prime: Exiting because mConverterError is 1650549857 (0x1000 req, 0x0 primed)
     AudioQueueObject.cpp:2480  Prime: failed (1650549857); will stop (4096/0 frames)
           AACDecoder.cpp:197    Too few bits left in input buffer
           AACDecoder.cpp:228    Error deserializing packet
  ACMP4AACBaseDecoder.cpp:1438  (0x11786d840) Error decoding packet 4: err = -1, packet length: 139
  ACMP4AACBaseDecoder.cpp:1447  '00 EA 38 09 01 F1 88 74 42 32 88 53 EE 83 59 59 6B DE 22 1E 00 28 14 A4 D0 11 75 98 0C 2A BF 6F DC 40 D3 3D 01 60 00 33 67 DE A2 21 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B 4B C0'
```

--- a/android/src/main/java/com/reactnativecompressor/Video/VideoCompressor/videoHelpers/Track.kt
+++ b/android/src/main/java/com/reactnativecompressor/Video/VideoCompressor/videoHelpers/Track.kt
@@ -258,10 +258,7 @@ class Track(id: Int, format: MediaFormat, audio: Boolean) {
     }
 
     private fun AudioSampleEntry.setup(format: MediaFormat): AudioSampleEntry = apply {
-        channelCount =
-            if (format.getInteger(MediaFormat.KEY_CHANNEL_COUNT) == 1) 2 else format.getInteger(
-                MediaFormat.KEY_CHANNEL_COUNT
-            )
+        channelCount = format.getInteger(MediaFormat.KEY_CHANNEL_COUNT)
         sampleRate = format.getInteger(MediaFormat.KEY_SAMPLE_RATE).toLong()
         dataReferenceIndex = 1
         sampleSize = 16

